cineast-core/src/main/java/org/vitrivr/cineast/core/features/Superpixel.java

```diff
@@ -0,0 +1,256 @@
+import org.vitrivr.cineast.core.color.RGBContainer;
+import org.vitrivr.cineast.core.data.segments.SegmentContainer;
+
+public class Superpixel {
+
+  // definition of superpixel algorithms
+  public static final int ALG_FH04 = 1; // Felzenszwalb-Huttenlocher
+  public static final int ALG_MS = 2; // MeanShift
+  public static final int ALG_SLIC = 3; // SLIC
+  public static final int ALG_WATERSHED = 4; // Watershed
+
+  // definition of image transformation
+  public static final int IMG_AVG = 1; // average frame
+  public static final int IMG_MED = 2; // median frame
+  public static final int IMG_REP = 3; // most representative frame
+
+  /**
+   * Code Inspiration Source: https://boofcv.org/index.php?title=Example_Superpixels Segments an
+   * image and calls visualization method
+   */
+  public static <T extends ImageBase<T>>
+  BufferedImage performSegmentation(ImageSuperpixels<T> alg, T color) {
+    GBlurImageOps.gaussian(color, color, 0.5, -1, null);
+    GrayS32 pixelToSegment = new GrayS32(color.width, color.height);
+    alg.segment(color, pixelToSegment);
+    return computation(pixelToSegment, color, alg.getTotalSuperpixels());
+  }
+
+  /**
+   * Code source: https://boofcv.org/index.php?title=Example_Superpixels
+   * <p>
+   * Don't need visualization, just computation of Buffered Image!
+   * <p>
+   */
+  public static <T extends ImageBase<T>>
+  BufferedImage computation(GrayS32 pixelToRegion, T color, int numSegments) {
+    // Computes the mean color inside each region
+    ImageType<T> type = color.getImageType();
+    ComputeRegionMeanColor<T> colorize = FactorySegmentationAlg.regionMeanColor(type);
+
+    FastQueue<float[]> segmentColor = new ColorQueue_F32(type.getNumBands());
+    segmentColor.resize(numSegments);
+
+    GrowQueue_I32 regionMemberCount = new GrowQueue_I32();
+    regionMemberCount.resize(numSegments);
+
+    ImageSegmentationOps.countRegionPixels(pixelToRegion, numSegments, regionMemberCount.data);
+    colorize.process(color, pixelToRegion, regionMemberCount, segmentColor);
+
+    // ATTENTION:
+    // When uncommenting the following line, the superpixels get colored according to the colors specified in ColorSuperpixel.java
+    // segmentColor = usePredefinedColorSet(segmentColor);
+
+    return VisualizeRegions.regionsColor(pixelToRegion, segmentColor, null);
+  }
+
+  /**
+   * Maps segment colors to predefined colors
+   *
+   * @param segmentColor mean segmentColor of different fields
+   * @return segmentColor according to the closest color in {@link org.vitrivr.cineast.core.features.ColorSuperpixel}
+   */
+  private static FastQueue<float[]> usePredefinedColorSet(FastQueue<float[]> segmentColor) {
+    ColorSuperpixel colorSuperpixel = new ColorSuperpixel();
+    for (int i = 0; i < segmentColor.size; i++) {
+      double minDeltaE = Double.MAX_VALUE;
+      RGBContainer minColor = null;
+      LabContainer c1 = ColorConverter.RGBtoLab((int) segmentColor.data[i][0],
```

Comment:
If all relevant operations are performed in Lab space anyway, why not just convert everything first, rather than using such a cumbersome representation?